### ISSUE TITLE
onData not triggered

### ISSUE THREAD

**Issue by NChief** [varspool/Wrench#61]
Using the basic EchoApplication:

``` php
$server->registerApplication('main', new \Wrench\Application\EchoApplication());
```

And some javascript/jQuery:

``` javascript
var connection = new WebSocket('wss://'+window.location.hostname+':7766/main');

connection.onmessage = function(e) {
  console.log('Server: ' + e.data);
}

$('#frm_send').submit(function(e) {
  connection.send($('#ws_text').val());
});
```

and html:

``` html
<form id="frm_send">
<input type="text" placeholder="send" id="ws_text"><input type="submit" value="send" id="ws_send" />
</form>
```

I do not receive the first data I send. when sending again, I receive what I sent the first time. Sending again I receive what I sent the second time and so on...

This seams like a bug where the onData function is not triggered at the right time or something.
I also see from the log that "Handling payload" is not triggering until the second time I send something, with what I sent the first time as input.

**Comment by emadruida** [comment]
In Socket.php, If I change line 312 from this
```
if ($this->firstRead == true && strlen($result) == 1) {
```
To this:
```
if (strlen($result) == 1) {
```
This problem is sovled for me and now it works via wss:// also
HTH

**Comment by dominics** [comment]
Thanks for the report.

Wow, weird. So if `strlen($result) == 1` we should be continuing to read bytes? I wonder what the value of `$length` in this context is.

**Comment by emadruida** [comment]
Well, I don't know if Chrome is still presenting this behaviour. I've tested with Firefox and looks like there are more bytes to be read if we use SSL

**Comment by leviathoskop** [comment]
@emadruida 

I am still facing problems with connecting via "wss".
Could you provide a little example of how you manage the secure websocket connection?
That would be a great help, because i'm struggeling with this problem for the last days.

**Comment by Alarmfifa** [comment]
@leviathoskop, We are using stunnel for proxy ssl connection.
I know, this is workaround, but it works =)

**Comment by emadruida** [comment]
@leviathoskop 
Once I've patched Socket.php, the only problem I faced was that I had to create a security exception in the client browser to accept the self-signed certificate. Unfortunately, the way to make this exception varies with the browser.
In Firefox, you have to navigate to the IP and Port of your web-socket server and accept the security warning:
So, navigate to `https://192.168.0.115:8082` and create the exception (note the https).
Beware, this will throw down your websocket server, so once accepted, go and launch it again.

HTH

**Comment by leviathoskop** [comment]
@emadruida 

This was very helpful, it solved the first problem that occured. 
But im Still facing another Problem:

Warning: stream_socket_accept(): SSL operation failed with code 1. OpenSSL Error messages:
error:14094412:SSL routines:SSL3_READ_BYTES:sslv3 alert bad certificate in X:\...\php_websockets_tests\Wrench-2.0.5\lib\Wrench\Socket\ServerSocket.php on line
87

Warning: stream_socket_accept(): Failed to enable crypto in X:\...\php_websockets_tests\Wrench-2.0.5\lib\Wrench\Socket\ServerSocket.php on line 87

Warning: stream_socket_accept(): accept failed: The operation completed successfully.
 in X:\...\php_websockets_tests\Wrench-2.0.5\lib\Wrench\Socket\ServerSocket.php on line 87

These are the errors that still occur .... has anybody an idea what i'm still doing wrong?

**Comment by emadruida** [comment]
@leviathoskop
I don't know if this is causing your issue, but the certificate served by the websocket server must have the private key concatenated to the certificate.

**Comment by emadruida** [comment]
@dominics
I've installed the new version and observe exactly the same behaviour as before. Whenever I send a message to the server, it is not handled until the next message arrives. This happens only when using WSS.

I've patched the Socket.php class as described above (now line 307) and then it works nicely both with WSS and with WS.

**Comment by kritter** [comment]
Can confirm the issue experienced by @emadruida. Strange problem o_O

**Comment by dominics** [comment]
The fix mentioned by @emadruida has been merged into v2.0.8 (and v3.0.x-dev) (9dd8175) - let me know if that fixes things
